gcm_toolkit/utils/interface.py

```diff
@@ -487,6 +487,156 @@ def phase_curve(self, phases, spectra=None, filename=None):
+            gravity = self.dsi.attrs.get(c["g"]) * 100
+
+        # check if radius is given, if not use gcm value
+        if rplanet is None:
+            rplanet = self.dsi.attrs.get(c["R_p"])
+
+        # check if pressure is given, if not use gcm value
+        if pressure_0 is None:
+            pressure_0 = np.max(pres)
+
+        # Avaraging over terminator regions in longitude space
+        ds_clouds_evening = self.dsi.where((self.dsi['lon'] > 90 - lon_resolution/2) *
+                                           (self.dsi['lon'] < 90 + lon_resolution/2),
+                                           drop=True).mean('lon')
+        ds_clouds_morning = self.dsi.where((self.dsi['lon'] > -90 - lon_resolution/2) *
+                                           (self.dsi['lon'] < -90 + lon_resolution/2),
+                                           drop=True).mean('lon')
+
+        # calculate latitude angle
+        lat_step = 180 / lat_points
+
+        # avarage over latitude space
+        spectra_list = []
+        for lat in range(lat_points):
+            tmp_morning = ds_clouds_morning.where((ds_clouds_evening['lat'] > lat*lat_step-90) *
+                                                  (ds_clouds_evening['lat'] < (lat+1)*lat_step-90),
+                                                  drop=True).mean('lat')
+            tmp_evening = ds_clouds_evening.where((ds_clouds_evening['lat'] > lat*lat_step-90) *
+                                                  (ds_clouds_evening['lat'] < (lat+1)*lat_step-90),
+                                                  drop=True).mean('lat')
+
+            # get temperature profile
+            temp = tmp_morning['T'].values[::-1]
+            # calculate chemistry in mass fractions
+            if mass_frac is None:
+                abus = interpol_abundances(co_ratios, feh_ratios, temp, pres)
+                for key in tlist:
+                    if key[0] in abus:
+                        abus[key[1]] = abus[key[0]]
+            else:
+                abus = mass_frac[0][lat]
+            # calcualte transmision spectra for the morning terminator of current lat point
+            self.prt.calc_transm(temp,
+                                 abus,
+                                 gravity,
+                                 mmw*np.ones_like(temp),
+                                 R_pl=rplanet*100,
+                                 P0_bar=pressure_0)
+            # add the spectra to the list
+            spectra_list.append(self.prt.transm_rad)
+
+            # get temperature profile
+            temp = tmp_evening['T'].values[::-1]
+            # calculate chemistry in mass fractions
+            if mass_frac is None:
+                abus = interpol_abundances(co_ratios, feh_ratios, temp, pres)
+                for key in tlist:
+                    if key[0] in abus:
+                        abus[key[1]] = abus[key[0]]
+            else:
+                abus = mass_frac[1][lat]
```

Comment:
I can see that the issue with the provided chemistry `self.chemistry` is that you can not use specific temperature values...  We would have to think about it...

Reply:
Yes! I'm happy to have a discussion about it.

Reply:
I will think about it!

Reply:
Hi, @Kiefersv, I think we can just add a Funktion to the chemistry module that lets you calculate the abundances for a specific pressure and temperature. It would essentially do the same as you do now. However, this is still more versatile in case we want to add more different chemistry schemes at some point.